crypto-jni/src/lib.rs

```diff
@@ -103,7 +100,13 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_verify(
         Ok(x) => x,
         _ => return NATIVE_CONVERSION_ERROR,
     };
-    let public_key = match PublicKey::from_bytes(&public_key_bytes) {
+
+    let public_key_bytes_array: [u8; 32] = match public_key_bytes.try_into() {
+        Ok(s) => s,
+        Err(_) => return MALFORMED_SECRET_KEY,
```

Comment:
This does not seem the right error.

Reply:
Whoops. Changed to `MALFORMED_PUBLIC_KEY`.